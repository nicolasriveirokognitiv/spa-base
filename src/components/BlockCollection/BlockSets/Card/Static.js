import React from 'react';
import PropTypes from 'prop-types';
import properties from './properties';
import { View, Tools } from '../../ElementCollection';
import { Button } from 'antd';
import getClassName from 'Utils/getClassName';

const { defaultValues } = Tools;

function Card({ config, data, design }) {
    return (
        <View.StyledWrapper
            className={getClassName({ imageLeft: config.flexDirection }, 'card-block')}
            config={config}
            design={design}
            testId={`${properties.type}-static`}
        >
            {data.blockTitle && (
                <div className="row">
                    <div className="col-12 text-align-center">
                        <div className="card-block-header">{data.blockTitle}</div>
                    </div>
                </div>
            )}
            <div className={`row flex-middle row-medium ${config.flexDirection || defaultValues.flexDirection}`}>
                <div className="col-6 col-sm-6">
                    <div className="card-block-title">
                        {data.title}
                    </div>
                    <div className="card-block-description">
                        {data.description}
                    </div>
                    <div className={`card-block-button ${config.buttonAlignment || defaultValues.textAlignment}`}>
                        <Button href="#" type="primary">{data.buttonLabel}</Button>
                    </div>
                </div>
                <div className="col-6 col-sm-6">
                    <div className="card-block-img-placeholder" />
                </div>
            </div>
        </View.StyledWrapper>
    );
}

Card.defaultProps = {
    config : {},
    design : {},
    data   : {},
};

Card.propTypes = {
    config: PropTypes.shape({
        flexDirection   : PropTypes.string,
        width           : PropTypes.string,
        buttonAlignment : PropTypes.string,
    }),
    design: PropTypes.shape({
        border       : PropTypes.shape({}),
        borderRadius : PropTypes.string,
    }),
    data: PropTypes.shape({
        title       : PropTypes.string,
        description : PropTypes.string,
        blockTitle  : PropTypes.string,
        buttonLabel : PropTypes.string,
    }),
};

export default Card;
