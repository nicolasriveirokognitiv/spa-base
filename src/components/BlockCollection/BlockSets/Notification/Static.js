import React from 'react';
import PropTypes from 'prop-types';
import properties from './properties';
import { View } from '../../ElementCollection';

function Notification({ config, data, design }) {
    return (
        <View.StyledWrapper
            className="notification-block"
            config={config}
            design={design}
            testId={`${properties.type}-static`}
        >
            <div className="notification-block-title">
                {data.title}
            </div>
            <div className="notification-block-description">
                {data.description}
            </div>
        </View.StyledWrapper>
    );
}

Notification.defaultProps = {
    config : {},
    design : {},
    data   : {},
};

Notification.propTypes = {
    config: PropTypes.shape({
        alignment : PropTypes.string,
        width     : PropTypes.string,
    }),
    design: PropTypes.shape({
        border       : PropTypes.shape({}),
        borderRadius : PropTypes.string,
    }),
    data: PropTypes.shape({
        title       : PropTypes.string,
        description : PropTypes.string,
    }),
};

export default Notification;
