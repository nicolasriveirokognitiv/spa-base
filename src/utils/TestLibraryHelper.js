/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
