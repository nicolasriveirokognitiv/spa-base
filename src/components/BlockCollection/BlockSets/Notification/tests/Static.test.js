import React from 'react';
import { render } from 'Utils/TestLibraryHelper';
import Static from '../Static';

describe('Notification Static', () => {
    test('Should render Correctly', () => {
        expect(<Static />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<Static />);
        expect(asFragment()).toMatchSnapshot();
    });
});
