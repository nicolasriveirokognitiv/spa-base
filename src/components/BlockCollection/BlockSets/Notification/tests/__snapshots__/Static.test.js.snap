// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Static Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-full"
  >
    <div
      class="styled-wrapper    notification-block"
    >
      <div
        class="notification-block-title"
      />
      <div
        class="notification-block-description"
      />
    </div>
  </div>
</DocumentFragment>
`;
